ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -4,20 +4,17 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
 CONFIG="${SHARED_DIR}/install-config.yaml"
 
 expiration_date=$(date -d '4 hours' --iso=minutes --utc)
 
 function join_by { local IFS="$1"; shift; echo "$*"; }
 
 REGION="${LEASED_RESOURCE}"
-case "${REGION}" in
-us-east-1)
-    ZONES=("us-east-1b" "us-east-1c")
-    ;;
-*)
-    ZONES=("${REGION}a" "${REGION}b")
-esac
+
+mapfile -t ZONES < <(aws --region "${REGION}" ec2 describe-availability-zones | jq -r '.AvailabilityZones[] | select(.State == "available") | .ZoneName'| sort -R)
 
 ZONES_COUNT=${ZONES_COUNT:-2}
```

Comment:
Interesting that the BYO VPC jobs default to 3 zones whereas the vanilla install defaults to 2 zones.

Reply:
Agreed

Reply:
One thing we should take in consideration is that using only 2 zones is not the default behavior a user would experience. It would be more representative to use 3 zones (default) unless it's not possible. Do you see any issue with me changing this?

Reply:
I think that we use 2 zones deliberately to reduce the resource load. Each availability zone requires a lot of additional resources (not the least of which is another elastic IP). With 2 AZs, we get coverage for multi-AZ will minimizing the resource usage.

Reply:
Understood. I will then mark this as resolved. Thanks for the clarification.